### PR TITLE
fix: Remove tag management page form, drop unused caching, fix None check

### LINKED ISSUE #6428: 'Save Changes' for deleting tags doesn't seem to work.
From support ticket: "Clicking the trash icon for each Tag fires a DELETE request with a 204 response, but the items remain in the list. When I click Save Changes (which sends a POST), there is a 405 METHOD NOT ALLOWED response and the page is blank."

I was able to eventually delete tags by repeating the process a couple times and refreshing the page but still not sure why it didn't work in the first place.

### PR DESCRIPTION
**Pull request by bretthoerner**
Fixes GH-6428

1. Fix broken falsey check on status (I forgot Python)
2. Removes old unused form on tag management page (must have been used for editing labels in the past).
3. Remove TagKey caching which was originally used in two places (before tagstore) which have been removed (as of #6433). Originally from: https://github.com/getsentry/sentry/blob/83a8334cae44b448a14e582be0365f04527b2f2c/src/sentry/models/tagkey.py#L40-L52

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/project_tags.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 class ProjectTagsEndpoint(ProjectEndpoint):
     def get(self, request, project):
-        tag_keys = tagstore.get_tag_keys(project.id)
+        tag_keys = sorted(tagstore.get_tag_keys(project.id), key=lambda x: x.key)
```
Comment (bretthoerner) [inline src/sentry/api/endpoints/project_tags.py]: I'm indifferent on whether we force Postgres to do the `ORDER BY` for us. Seemed better to let callers just sort the keys?

Comment (mattrobenolt) [reply src/sentry/api/endpoints/project_tags.py]: Yeah, I think this is fine for now. We don’t have a ton of stuff to sort. And less work for the databass the better imo.

**Review by mattrobenolt — APPROVED** [review]
:pizza: 🥇

**Pull request merged**